pkgs/applications/audio/soundconverter/default.nix

```diff
@@ -0,0 +1,80 @@
+{ lib, fetchurl
+# Optional due to unfree license.
+, faacSupport ? false
+, python3Packages, gtk3, wrapGAppsHook
+, gsettings-desktop-schemas, intltool, xvfb-run
+, gobject-introspection, gst_all_1, fdk-aac-encoder }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "soundconverter";
+  version = "4.0.3";
+
+  src = fetchurl {
+    url = "https://launchpad.net/soundconverter/trunk/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "sha256-hzIG/4LD3705erPYvXb7uoRwF9LtKKIKB3jrhpYMsZ0=";
+  };
+
+  buildInputs = [
+    gtk3
+    wrapGAppsHook
+  ];
+
+  nativeBuildInputs = [
+    intltool
+    gobject-introspection
+  ];
+
+  propagatedBuildInputs = [
+    gobject-introspection
+    fdk-aac-encoder
+    python3Packages.gst-python
+    python3Packages.distutils_extra
+    python3Packages.setuptools
+    gst_all_1.gst-libav
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+    gst_all_1.gst-plugins-ugly
+    (gst_all_1.gst-plugins-bad.override { inherit faacSupport; })
+  ];
+
+  checkInputs = [
+    xvfb-run
+  ];
+
+  preCheck = let
+    gds = gsettings-desktop-schemas;
+  in ''
+    export HOME=$TMPDIR
+    addToSearchPath XDG_DATA_DIRS "${gds}/share/gsettings-schemas/${gds.name}"
+    addToSearchPath XDG_DATA_DIRS "${gtk3}/share/gsettings-schemas/${gtk3.name}"
+    addToSearchPath XDG_DATA_DIRS "$out/share/gsettings-schemas/${pname}-${version}"
+  '' + lib.optionalString (!faacSupport) ''
+    substituteInPlace  tests/testcases/integration.py --replace \
+      "for encoder in ['fdkaacenc', 'faac', 'avenc_aac']:" \
+      "for encoder in ['fdkaacenc', 'avenc_aac']:"
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+    xvfb-run python tests/test.py
+  '';
+
+  postPatch = ''
```

Comment:
It might be nicer to move this before `checkPhase` since it occurs sooner:

https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md

Reply:
Sure.